### ISSUE TITLE
Changelog ?

### ISSUE THREAD

**Issue by adrien-may** [opentok/web-components#20]
Hi there,
I'm deeply sorry if this information is available elsewhere but I failed to find it.

Could we add it in a dedicated `CHANGELOG.md`, in the `README.md`or via github tags. ?

I'd like to update my package to the latest version but I need to ensure no breaking changes occurred. Thanks in advance

**Comment by conshus** [comment]
Hi @adrien-may!

That's a really good idea! I'll create a CHANGELOG.md in the next update.
Right now, we are in the "get people to try it out and give feedback" stage of development. 
There haven't been any breaking changes. Mostly adding things people suggest.

What version are you using now?
How are you finding using the Web Components so far?

**Comment by adrien-may** [comment]
Using 0.0.3 right now. Working ok so far, we’ve been using the react
deprecated sdk for quite a while, this web component approach seems to work
better !

Le mar. 1 août 2023 à 18:11, Dwane Hemmings ***@***.***> a
écrit :

**Comment by conshus** [comment]
@adrien-may 
Sounds good. As you can see we are still in v0.0.x and looking for people's feedback before we publish v1.0.0 with everything stable.

I was just reminded from another issue that we went from setting the width and height on the elements, i.e.:

```
 <video-publisher witdh="320px" height="240px"></video-publisher>
```

to setting a properties object (see: https://tokbox.com/developer/sdks/js/reference/OT.html#initPublisher) because folks said they wanted to have more properties to change.

So if you wanted to set the properties, it would be the same as setting the session and token.

```
publisher.current.properties = {width: '320px', height: '240px'}
```

I'm going to update the sample docs and code to be more explicit.

**Comment by adrien-may** [comment]
Hi there, we're still on the same old version because we're not quite sure if we can safely migrate to the latest versions. Is a changelog going to be commited soon ?

**Comment by conshus** [comment]
Hi @adrien-may 
Nothing has changed since your message except some documentation updates. When you say the old version, do you mean you are setting the height and width like:
` <video-publisher witdh="320px" height="240px"></video-publisher>`
instead setting the properties like
`publisher.current.properties = {width: '320px', height: '240px'}`

I don't see anything major changing for a while. We've just released a beta of Video API to be used with Vonage applications for a more unified experience with everything in one dashboard.

As for the Web Components, nothing functionally will change. The only difference will be in the way you generate the session and token. So regardless if you are still using OpenTok creds or Vonage creds, you just need to pass in the session and token for the Web Components to work just like normal.

I'm going to be converting and creating some applications using the Web Components with Vonage credentials to make sure everything is the same and will start a change log announcing being able to use Vonage credentials with the current versions of the components and update the examples and docs.

In conclusion, aside from testing using Vonage credentials. I don't see the Web Components changing (functionality) for a while. I'm still trying to get more feedback before anything will change. If it's working for folks, we'll leave it as it is for the foreseeable future and release a v1 early next year.

Hope that helps. Let me know if you need anything more.

How do you find using the Web Components in your application? Any feedback?

**Comment by adrien-may** [comment]
By old version, i mean 0.0.3 and yes I'm still using ` <video-publisher witdh="320px" height="240px"></video-publisher>`

My point is : I don't want to and wont in any case update a software from one version to another without understanding what's at stake. That's the whole point of a changelog :)

If there is no breaking changes, that's great news but without reading the full code diff, I can't get to that conclusion.

Components seems to work ok so far. We still have to investiguate one issue on our side with the video being cut after a few minutes but I don't think it's directly linked to this package.

Cheers

**Comment by conshus** [comment]
Got it. Makes sense. Be on the lookout for a changelog in the new year after I make sure the unified integration also works and update the examples.

> We still have to investiguate one issue on our side with the video being cut after a few minutes but I don't think it's directly linked to this package.

So, the video feed just turns off or does the call end? Have you tried using the example in the repo to make sure everything works?

Thanks.
